transcoder.go

```diff
@@ -216,30 +218,43 @@ func (t *Transcoder) registerMethod(handler http.Handler, methodDesc protoreflec
 		methodConf.streamType = connect.StreamTypeUnary
 	}
 
+	// Add the implicit HTTP POST route for the method.
+	defaultTarget, err := t.restRoutes.addDefaultRoute(methodConf)
+	if err != nil {
+		return fmt.Errorf("failed to add default REST route for method %s: %w", methodPath, err)
+	}
+	methodConf.restDefaultTarget = defaultTarget
+	// Add the explicit declared HTTP rules for the method.
 	if httpRule, ok := getHTTPRuleExtension(methodDesc); ok {
-		if err := t.addRule(httpRule, methodConf); err != nil {
+		targets, err := t.addRule(httpRule, methodConf)
+		if err != nil {
 			return err
 		}
+		methodConf.restTargets = targets
 	}
 	return nil
 }
 
-func (t *Transcoder) addRule(httpRule *annotations.HttpRule, methodConf *methodConfig) error {
+// addRule adds an HTTP rule to the transcoder, creating a route for each binding.
+func (t *Transcoder) addRule(httpRule *annotations.HttpRule, methodConf *methodConfig) ([]*routeTarget, error) {
+	targets := make([]*routeTarget, 0, 1+len(httpRule.GetAdditionalBindings()))
 	methodPath := methodConf.methodPath
 	firstTarget, err := t.restRoutes.addRoute(methodConf, httpRule)
 	if err != nil {
-		return fmt.Errorf("failed to add REST route for method %s: %w", methodPath, err)
+		return nil, fmt.Errorf("failed to add REST route for method %s: %w", methodPath, err)
 	}
-	methodConf.httpRule = firstTarget
```

Comment:
Removed the assignment here to make it visible to the caller that we modify methodConf. Previously this was always taking any `WithRules` to be the last write and therefore the "first" rule, but I don't believe that's intentional. So now we append to the list of targets to also make it easier to debug how many routes are registered for a single method.